src/modules/category/dto/received/create-category.dto.ts

```diff
@@ -0,0 +1,12 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsArray, IsNotEmpty } from 'class-validator';
+import { Type } from 'class-transformer';
+import { CategoryDto } from '../../../database/dto/category.dto';
+
+export class CreateCategoryDto {
+    @ApiProperty({ description: 'new category entity', example: CategoryDto })
+    @IsNotEmpty()
+    @IsArray()
+    @Type(() => CategoryDto)
+    category: CategoryDto
```

Comment:
type instead example, and is it an array?

Reply:
done